custom_components/deltasol/sensor.py

```diff
@@ -46,6 +47,22 @@
     }
 )
 
+async def update_unique_ids(hass, data):
+    _LOGGER.debug("Checking for sensors with old ids in registry.")
+    ent_reg = await entity_registry.async_get_registry(hass)
+
+    for unique_id, endpoint in data.items():
+        name_id = ent_reg.async_get_entity_id("sensor", DOMAIN, endpoint.name)
+        if name_id is not None:
+            _LOGGER.info(f"Found entity with old id ({name_id}). Updating to new unique_id ({unique_id}).")
+            # check if there already is a new one
+            new_entity_id = ent_reg.async_get_entity_id("sensor", DOMAIN, unique_id)
+            if new_entity_id is not None:
+                _LOGGER.info("Found entity with old id and an entity with a new unique_id. Preserving old enity...")
+                ent_reg.async_remove(new_entity_id)
+            ent_reg.async_update_entity(name_id, new_unique_id=unique_id)
```

Comment:
Is new_unique_id=unique_id saying that it should take the var  new_unique_id for the property unique_id?

Reply:
The other way around:
https://github.com/home-assistant/core/blob/865159781fe24b297526dbc293fa9cfbcc719ebb/homeassistant/helpers/entity_registry.py#L511